WeakAuras/RegionTypes/DynamicGroup.lua

```diff
@@ -810,106 +818,145 @@ local function modify(parent, region, data)
-      controlPoint:ClearAnchorPoint()
-      controlPoint:SetAnchorPoint(data.selfPoint, self, data.selfPoint, x, y)
-      controlPoint:SetShown(show)
-      controlPoint:SetWidth(regionData.data.width or regionData.region.width)
-      controlPoint:SetHeight(regionData.data.height or regionData.region.height)
-      if animate then
-        WeakAuras.CancelAnimation(regionData.controlPoint, true)
-        local xPrev = regionData.xOffset or x
-        local yPrev = regionData.yOffset or y
-        local xDelta = xPrev - x
-        local yDelta = yPrev - y
-        if show and (abs(xDelta) > 0.01 or abs(yDelta) > 0.01) then
-          local anim
-          if data.grow == "CIRCLE" or data.grow == "COUNTERCIRCLE" then
-            local originX, originY = 0,0
-            local radius1, previousAngle = WeakAuras.GetPolarCoordinates(xPrev, yPrev, originX, originY)
-            local radius2, newAngle = WeakAuras.GetPolarCoordinates(x, y, originX, originY)
-            local dAngle = newAngle - previousAngle
-            dAngle = ((dAngle > 180 and dAngle - 360) or (dAngle < -180 and dAngle + 360) or dAngle)
-            if(math.abs(radius1 - radius2) > 0.1) then
-              local translateFunc = [[
-                                function(progress, _, _, previousAngle, dAngle)
-                                    local previousRadius, dRadius = %f, %f;
-                                    local targetX, targetY = %f, %f
-                                    local radius = previousRadius + (1 - progress) * dRadius;
-                                    local angle = previousAngle + (1 - progress) * dAngle;
-                                    return cos(angle) * radius - targetX, sin(angle) * radius - targetY;
-                                end
-                            ]]
-              anim = {
-                type = "custom",
-                duration = 0.2,
-                use_translate = true,
-                translateType = "custom",
-                translateFunc = translateFunc:format(radius1, radius2 - radius1, x, y),
-                x = previousAngle,
-                y = dAngle,
-                selfPoint = data.selfPoint,
-                anchor = self,
-                anchorPoint = data.selfPoint,
-              }
-            else
-              local translateFunc = [[
-                                function(progress, _, _, previousAngle, dAngle)
-                                    local radius = %f;
-                                    local targetX, targetY = %f, %f
-                                    local angle = previousAngle + (1 - progress) * dAngle;
-                                    return cos(angle) * radius - targetX, sin(angle) * radius - targetY;
-                                end
-                            ]]
+    local anchorPerUnit = data.anchorFrameType == "NAMEPLATE" or data.anchorFrameType == "UNITFRAME"
+    local function positionSortedChildren(pos, children)
```

Comment:
DoPositionChildren is called a lot. I don’t think it is a good idea to recreate this inner function every time.